flytekit/core/tracker.py

```diff
@@ -92,6 +137,29 @@ def find_lhs(self) -> str:
                 # continue looping through m.
                 logger.warning("Caught ValueError {} while attempting to auto-assign name".format(err))
 
+        # try to find object in module when the tracked instance is defined in the __main__ module
+        module = import_module_from_file(self._instantiated_in, self._module_file)
+
+        def _candidate_name_matches(candidate) -> bool:
+            if not hasattr(candidate, "name") or not hasattr(self, "name"):
```

Comment:
can we add a comment as to when is this the case?

Reply:
not all the attributes in the module will have a `name` attr